plugins/by-name/dap/default.nix

```diff
@@ -62,32 +55,32 @@ with dapHelpers;
     };
   };
 
-  config =
+  # Separate configuration and adapter configurations
+  callSetup = false;
+  extraConfig =
+    cfg:
     let
-      options =
-        with cfg;
-        {
-          inherit configurations;
+      options = {
+        inherit (cfg) configurations;
 
-          adapters =
-            (lib.optionalAttrs (adapters.executables != null) (
-              processAdapters "executable" adapters.executables
-            ))
-            // (lib.optionalAttrs (adapters.servers != null) (processAdapters "server" adapters.servers));
+        adapters =
+          (lib.optionalAttrs (cfg.adapters.executables != null) (
+            dapHelpers.processAdapters "executable" cfg.adapters.executables
+          ))
+          // (lib.optionalAttrs (cfg.adapters.servers != null) (
+            dapHelpers.processAdapters "server" cfg.adapters.servers
+          ));
 
-          signs = with signs; {
-            DapBreakpoint = dapBreakpoint;
-            DapBreakpointCondition = dapBreakpointCondition;
-            DapLogPoint = dapLogPoint;
-            DapStopped = dapStopped;
-            DapBreakpointRejected = dapBreakpointRejected;
-          };
-        }
-        // cfg.extraOptions;
+        signs = with cfg.signs; {
+          DapBreakpoint = dapBreakpoint;
+          DapBreakpointCondition = dapBreakpointCondition;
+          DapLogPoint = dapLogPoint;
+          DapStopped = dapStopped;
+          DapBreakpointRejected = dapBreakpointRejected;
+        };
+      } // cfg.settings;
     in
-    mkIf cfg.enable {
-      extraPlugins = [ cfg.package ];
-
+    {
       extraConfigLua =
         (optionalString (cfg.adapters != null) ''
           require("dap").adapters = ${lib.nixvim.toLuaObject options.adapters}
```

Comment:
Using `optionalString` here for what should be an optional _definition_ is means we will always add at least one line to extraConfigLua.

Not a big issue, but we've had complaints from OCD users in the past :grin: 

Ideally, we'd use `mkMerge` and `mkIf` here, unless that proves ridiculously ugly in practice.

Also: the current impl does not need the parenthesis for string concatenation.

Also: should we wrap this in a `do` block?

Reply:
Updated it, let me know what ya think. I think the generated config looks good.

Reply:
I was thinking 

```nix
{
  extraConfigLua = mkMerge [
    (mkIf true "")
    (mkIf false "")
    # ...
  ];
}
```

?

Reply:
I thought I did that it complained.. redid it and it worked.